src/server/server.rs

```diff
@@ -344,6 +396,35 @@ where
         Ok(())
     }
 
+    pub fn pause(&mut self) -> Result<()> {
```

Comment:
is there any test case to cover pause and resume logic?

Reply:
Supplied.